### PR TITLE
Fix holodeck tables being invisible and chairs facing the wrong way

### PR DESCRIPTION
**Pull request by CinnamonSnowball**
## What Does This PR Do
- Replaces the janky "spawn in nullspace then directly overwrite `loc`" solution in `/area/proc/copy_contents_to()` with a parameter the proc already had, `atom/newloc`. This means the loc is set earlier in the duplication code, and lets `.update_icon()` run on an actual turf rather than nullspace.
For tables specifically, this means they won't be invisible in the holodeck anymore, as they can finally smooth properly.
- Removes the explicit `perfect_copy = FALSE` argument pass in holodeck code, which means the copied stools and chairs will inherit the facing direction properly.
- Cleans up a bunch of obvious shitcode throughout the code file. Didn't touch the more complicated parts as I was less sure it'd break something.

## Why It's Good For The Game
Invisible objects bad, chairs facing the correct way good

## Changelog
:cl:
fix: Holodeck tables are no longer invisible
fix: Holodeck chairs now face the correct direction
/:cl:

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -705,7 +705,7 @@ Returns 1 if the chain up to the area contains the given typepath
 		var/area/areatemp = areatype
 		areatype = areatemp.type
 
-	var/list/areas = new/list()
+	var/list/areas = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:708]:
```suggestion
	var/list/areas = list()
```
standard syntax

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -719,7 +719,7 @@ Returns 1 if the chain up to the area contains the given typepath
 		var/area/areatemp = areatype
 		areatype = areatemp.type
 
-	var/list/turfs = new/list()
+	var/list/turfs = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:722]:
```suggestion
	var/list/turfs = list()
```

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -770,24 +770,24 @@ Returns 1 if the chain up to the area contains the given typepath
 		if(T.x < trg_min_x || !trg_min_x) trg_min_x	= T.x
 		if(T.y < trg_min_y || !trg_min_y) trg_min_y	= T.y
 
-	var/list/refined_src = new/list()
+	var/list/refined_src = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:773]:
```suggestion
	var/list/refined_src = list()
```

File: code/__HELPERS/unsorted.dm [hunk 4]
```diff
@@ -770,24 +770,24 @@ Returns 1 if the chain up to the area contains the given typepath
...
 	for(var/turf/T in turfs_src)
 		refined_src += T
 		refined_src[T] = new/datum/coords
 		var/datum/coords/C = refined_src[T]
 		C.x_pos = (T.x - src_min_x)
 		C.y_pos = (T.y - src_min_y)
 
-	var/list/refined_trg = new/list()
+	var/list/refined_trg = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:781]:
```suggestion
	var/list/refined_trg = list()
```

File: code/__HELPERS/unsorted.dm [hunk 5]
```diff
@@ -770,24 +770,24 @@ Returns 1 if the chain up to the area contains the given typepath
...
 		refined_trg += T
 		refined_trg[T] = new/datum/coords
 		var/datum/coords/C = refined_trg[T]
 		C.x_pos = (T.x - trg_min_x)
 		C.y_pos = (T.y - trg_min_y)
 
-	var/list/fromupdate = new/list()
-	var/list/toupdate = new/list()
+	var/list/from_update = new
+	var/list/to_update = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:790]:
```suggestion
	var/list/from_update = list()
	var/list/to_update = list()
```

File: code/__HELPERS/unsorted.dm [hunk 6]
```diff
@@ -941,103 +939,75 @@ Returns 1 if the chain up to the area contains the given typepath
 		if(T.y < trg_min_y || !trg_min_y)
 			trg_min_y	= T.y
 
-	var/list/refined_src = new/list()
+	var/list/refined_src = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:942]:
```suggestion
	var/list/refined_src = list()
```

File: code/__HELPERS/unsorted.dm [hunk 7]
```diff
@@ -941,103 +939,75 @@ Returns 1 if the chain up to the area contains the given typepath
...
 	for(var/turf/T in turfs_src)
 		refined_src += T
 		refined_src[T] = new/datum/coords
 		var/datum/coords/C = refined_src[T]
 		C.x_pos = (T.x - src_min_x)
 		C.y_pos = (T.y - src_min_y)
 
-	var/list/refined_trg = new/list()
+	var/list/refined_trg = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:950]:
```suggestion
	var/list/refined_trg = list()
```

File: code/__HELPERS/unsorted.dm [hunk 8]
```diff
@@ -941,103 +939,75 @@ Returns 1 if the chain up to the area contains the given typepath
...
 		var/datum/coords/C = refined_trg[T]
 		C.x_pos = (T.x - trg_min_x)
 		C.y_pos = (T.y - trg_min_y)
 
-	var/list/toupdate = new/list()
-
-	var/copiedobjs = list()
-
+	var/list/to_update = new
+	var/list/copied_objects = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:959]:
```suggestion
	var/list/to_update = list()
	var/list/copied_objects = list()
```

File: code/__HELPERS/unsorted.dm [hunk 9]
```diff
@@ -941,103 +939,75 @@ Returns 1 if the chain up to the area contains the given typepath
...
 					X.icon_state = old_icon_state1
-					X.icon = old_icon1 //Shuttle floors are in shuttle.dmi while the defaults are floors.dmi
+					X.icon = old_icon1 // Shuttle floors are in shuttle.dmi while the defaults are floors.dmi
 
-					var/list/objs = new/list()
-					var/list/newobjs = new/list()
-					var/list/mobs = new/list()
-					var/list/newmobs = new/list()
+					var/list/newobjs = new
+					var/list/newmobs = new
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:982]:
```suggestion
					var/list/newobjs = list()
					var/list/newmobs = list()
```

File: code/__HELPERS/unsorted.dm [hunk 10]
```diff
@@ -941,103 +939,75 @@ Returns 1 if the chain up to the area contains the given typepath
...
-					var/list/newobjs = new/list()
-					var/list/mobs = new/list()
-					var/list/newmobs = new/list()
+					var/list/newobjs = list()
+					var/list/newmobs = list()
 
 					for(var/obj/O in T)
-
-						if(!isobj(O))
-							continue
```
Comment (Sirryan2002) [inline code/__HELPERS/unsorted.dm:992]: old coders were surely on some drug

**Review by Sirryan2002 — APPROVED** [review]
looks good, nice work

File: code/__HELPERS/unsorted.dm [hunk 11]
```diff
@@ -1693,11 +1663,11 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 	if(!isnull(value) && value != "")
 		matches = filter_fancy_list(matches, value)
 
-	if(matches.len == 0)
+	if(length(matches) == 0)
```
Comment (lewcc) [inline code/__HELPERS/unsorted.dm:1666]:
```suggestion
	if(!length(matches))
```

File: code/__HELPERS/unsorted.dm [hunk 12]
```diff
@@ -1385,9 +1355,9 @@ Standard way to write links -Sayu
 	var/list/processing_list = list(src)
 	var/list/processed = list()
 
-	var/atom/found = null
+	var/atom/found
 
-	while(processing_list.len && found==null)
+	while(length(processing_list) && found == null)
```
Comment (lewcc) [inline code/__HELPERS/unsorted.dm:1360]:
```suggestion
	while(length(processing_list) && isnull(found))
```

**Review by DGamerL — COMMENTED** [review]
While we are here

**Review by S34NW — CHANGES_REQUESTED** [review]
Please address reviews

File: code/__HELPERS/unsorted.dm [hunk 13]
```diff
@@ -834,20 +832,22 @@ Returns 1 if the chain up to the area contains the given typepath
...
 							X.icon = 'icons/turf/shuttle.dmi'
 							X.icon_state = replacetext(O.icon_state, "_f", "_s") // revert the turf to the old icon_state
 							X.name = "wall"
 							qdel(O) // prevents multiple shuttle corners from stacking
 							continue
-						if(!isobj(O)) continue
+
+						if(!isobj(O))
+							continue
+
```
[the remark this answers is not in the document]
Comment (CinnamonSnowball) [reply code/__HELPERS/unsorted.dm:846]: There's a conditional qdel call above, so I think it's valid to assume `O` might not exist anymore?

[the remark this answers is not in the document]
Comment (Contrabang) [reply code/__HELPERS/unsorted.dm:846]: Use QDELETED(O) instead imo

[the remark this answers is not in the document]
Comment (DGamerL) [reply code/__HELPERS/unsorted.dm:846]: Slightly too late, but yes, that is better

**Review by S34NW — APPROVED** [review]

**Pull request merged**